### PR TITLE
Implement rearrange_dim

### PR DESCRIPTION
**Pull request by yamaguchi1024**
This is part of `lift_alloc` cleanup

### CODE REVIEW

**Comment by yamaguchi1024** [comment]
ping @alexreinking @gilbo

File: SYS_ATL/API.py
Comment (alexreinking) [inline SYS_ATL/API.py:1054]: OdDlY forMaTteD **_lines_** like **this** ᴅɪsᴛʀᴀᴄᴛ readers/**reviewers** a n d **eat** up _time_.

1. The formatting of this function is internally inconsistent:
    1. Two spaces after `loopir` and before `=`?
    2. Spaces inside a function call? `f( x )` vs `f(x)`?
2. My review time is better spent discussing program logic than whitespace.
3. **Your engineering time is better spent anywhere besides fiddling with whitespace.**

On Monday, I want to re-raise the issue of mechanizing our code formatting, ref. #59

File: SYS_ATL/API.py
Comment (alexreinking) [inline SYS_ATL/API.py:1045]: Multi-target test, please.

File: tests/test_schedules.py
Comment (alexreinking) [inline tests/test_schedules.py:20]: _Check_ this.

File: tests/test_schedules.py
Comment (alexreinking) [inline tests/test_schedules.py:13]: Another point on formatting: `[N, M, K]` or `[N,M,K]`?

File: SYS_ATL/API.py
Comment (alexreinking) [inline SYS_ATL/API.py:1051]: No need for `+`.

**Review by alexreinking — CHANGES_REQUESTED** [review]

File: SYS_ATL/API.py
Comment (yamaguchi1024) [reply SYS_ATL/API.py:1054]: I get you were frustrated but this comment was quite funny to me

File: tests/test_schedules.py
Comment (yamaguchi1024) [reply tests/test_schedules.py:13]: pretty printing is `N, M, K`, so the former is more consistent?

File: SYS_ATL/API.py
Comment (alexreinking) [reply SYS_ATL/API.py]: I _did_ try to cut my frustration with a good bit of humor 🙂

**Comment by yamaguchi1024** [comment]
There is a bug, trying to fix. Please don't merge yet

**Comment by alexreinking** [comment]
> There is a bug, trying to fix. Please don't merge yet

It won't auto-merge until CI passes _and_ I approve, so no risk there.

**Comment by yamaguchi1024** [comment]
CI seems to be failing when installing clang..?

File: SYS_ATL/LoopIR_scheduling.py
Comment (alexreinking) [inline SYS_ATL/LoopIR_scheduling.py:1232]: Just a note: it would be really nice if we had a method to update a subset of records in an AST node... something like:

```
s.with(idx=new_idx)
```

File: SYS_ATL/LoopIR_scheduling.py
Comment (alexreinking) [inline SYS_ATL/LoopIR_scheduling.py:1241]: Imagine: `e.with(idx=new_idx)` or even `e.with(idx=[e.idx[i] for i in self.dimensions])`.

File: SYS_ATL/LoopIR_scheduling.py
Comment (alexreinking) [inline SYS_ATL/LoopIR_scheduling.py:1225]: Imagine:
```
return [s.with(type=s.type.with(hi=[s.type.hi[i] for i in self.dimensions]))]
```
Or imagine adding a helper function to this class:
```
def _permute(self, xs):
    return [xs[i] for i in self.dimensions]
```
Then it could just be:
```
return [s.with(type=s.type.with(hi=self._permute(s.type.hi)))]
```

**Review by alexreinking — APPROVED** [review]
LGTM now... comments are just musing/suggestions for the future.

**Pull request merged**
